urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/logica/ia/evaluacion/EvaluacionEstrategica.java

```diff
@@ -0,0 +1,223 @@
+	}
+	
+
+	private double porcentajeJugadasPosiblesRival(Tablero tablero, EstadoCasilla colorJugador) {
+		EstadoCasilla colorRival = colorJugador == EstadoCasilla.BLACK ? EstadoCasilla.WHITE : EstadoCasilla.BLACK;
+		int movidasPosiblesRival=0;
+		
+		Set<Point> movidasPosibles = ExploradorMovimientos.explorar(tablero, colorJugador);
+		if (movidasPosibles.isEmpty()) { /* se pierde turno - revisar siguiente jugador */
+			movidasPosibles = ExploradorMovimientos.explorar(tablero, colorRival);
+			if (movidasPosibles.isEmpty()) { // fin del juego
+					return 0;
+			} else {
+				movidasPosiblesRival = movidasPosibles.size();
+			}
+		} else { /* revisar el puntaje después de una movida */
+			for (Point siguienteMovimientoPosible : movidasPosibles) {
+				Tablero subTablero = tablero.clone();
+				if (siguienteMovimientoPosible != null) {
+					subTablero.hacerMovimiento(siguienteMovimientoPosible, colorJugador);
+					movidasPosibles = ExploradorMovimientos.explorar(subTablero, colorRival);
+					if (movidasPosibles.isEmpty()) { // fin del juego
+						return 0;
+					} else if(movidasPosibles.size() > movidasPosiblesRival){/*toma el peor de los casos (la mayor cantidad de libertades para el oponente)*/
+						movidasPosiblesRival = movidasPosibles.size();
+					}
+				}
+				
+			}
+		}
+		return -(double)movidasPosiblesRival/(double)Tablero.getCasillasjugablesIniciales();  /*  /(double)totalCasillasJugables */
+	}
+
+	private double porcentajeCasillasCentrales(Tablero tablero, EstadoCasilla colorJugador) {
+		int totalCasillasCentrales = 4;
+		int casillasCentralesJugador=0;
+		int casillasCentralesRival=0;
+		EstadoCasilla colorRival = colorJugador == EstadoCasilla.BLACK ? EstadoCasilla.WHITE : EstadoCasilla.BLACK;
+		List<Point> centrales = new ArrayList<Point>(4);
+		centrales.add(new Point((int)(Tablero.TABLERO_LARGO/2)-1, (int)(Tablero.TABLERO_LARGO/2)-1));
+		centrales.add(new Point((int)(Tablero.TABLERO_LARGO/2)-1, (int)(Tablero.TABLERO_LARGO/2)));
+		centrales.add(new Point((int)(Tablero.TABLERO_LARGO/2), (int)(Tablero.TABLERO_LARGO/2)-1));
+		centrales.add(new Point((int)(Tablero.TABLERO_LARGO/2), (int)(Tablero.TABLERO_LARGO/2)));
+		ListIterator<Point> iterador = centrales.listIterator();
+		while( iterador.hasNext() ) {
+		       Point punto = (Point) iterador.next();
+				if (tablero.obtenerTablero().get(punto) == colorJugador){
+					casillasCentralesJugador++;
+				}else if (tablero.obtenerTablero().get(punto) == colorRival) {
+					casillasCentralesRival++;
+				}
+		}
+		
+		/*System.out.println(colorJugador +" = " + casillasCentralesJugador);
+		System.out.println(colorRival +" = " + casillasCentralesRival);
+		System.out.println((double)(casillasCentralesJugador-casillasCentralesRival)/(double)totalCasillasCentrales);*/
+		return (double)(casillasCentralesJugador-casillasCentralesRival)/(double)totalCasillasCentrales;
+	}
+	
+
+	private List<Point> detectarEsquinas(Tablero tablero, TipoTablero tipoTablero) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 169-182 into a separate method.